resolver/init.sls

```diff
@@ -21,6 +21,7 @@ resolv-file:
         nameservers: {{ salt['pillar.get']('resolver:nameservers', ['8.8.8.8','8.8.4.4']) }}
         searchpaths: {{ salt['pillar.get']('resolver:searchpaths', [salt['grains.get']('domain'),]) }}
         options: {{ salt['pillar.get']('resolver:options', []) }}
+        domain: {{ salt['pillar.get']('resolver:domain') }}
```

Comment:
It's an optional parameter, must be explicitly set in pillar.
Setting a default will result in unwanted client behaviour.